Common/Securities/Option/QLOptionPriceModel.cs

```diff
@@ -0,0 +1,227 @@
+        /// <param name="dividendYieldEstimator">The underlying dividend yield estimator</param>
+        public QLOptionPriceModel(PricingEngineFunc pricingEngineFunc, IQLUnderlyingVolatilityEstimator underlyingVolEstimator, IQLRiskFreeRateEstimator riskFreeRateEstimator, IQLDividendYieldEstimator dividendYieldEstimator)
+        {
+            _pricingEngineFunc = (option, process) => pricingEngineFunc(process);
+            _underlyingVolEstimator = underlyingVolEstimator ?? new DefaultQLUnderlyingVolatilityEstimator();
+            _riskFreeRateEstimator = riskFreeRateEstimator ?? new DefaultQLRiskFreeRateEstimator();
+            _dividendYieldEstimator = dividendYieldEstimator ?? new DefaultQLDividendYieldEstimator();
+        }
+        /// <summary>
+        /// Method constructs QuantLib option price model with necessary estimators of underlying volatility, risk free rate, and underlying dividend yield
+        /// </summary>
+        /// <param name="pricingEngineFunc">Function takes option and modeled stochastic process, and returns new pricing engine to run calculations for that option</param>
+        /// <param name="underlyingVolEstimator">The underlying volatility estimator</param>
+        /// <param name="riskFreeRateEstimator">The risk free rate estimator</param>
+        /// <param name="dividendYieldEstimator">The underlying dividend yield estimator</param>
+        public QLOptionPriceModel(PricingEngineFuncEx pricingEngineFunc, IQLUnderlyingVolatilityEstimator underlyingVolEstimator, IQLRiskFreeRateEstimator riskFreeRateEstimator, IQLDividendYieldEstimator dividendYieldEstimator)
+        {
+            _pricingEngineFunc = pricingEngineFunc;
+            _underlyingVolEstimator = underlyingVolEstimator ?? new DefaultQLUnderlyingVolatilityEstimator();
+            _riskFreeRateEstimator = riskFreeRateEstimator ?? new DefaultQLRiskFreeRateEstimator();
+            _dividendYieldEstimator = dividendYieldEstimator ?? new DefaultQLDividendYieldEstimator();
+        }
+
+        /// <summary>
+        /// Evaluates the specified option contract to compute a theoretical price, IV and greeks
+        /// </summary>
+        /// <param name="security">The option security object</param>
+        /// <param name="slice">The current data slice. This can be used to access other information
+        /// available to the algorithm</param>
+        /// <param name="contract">The option contract to evaluate</param>
+        /// <returns>An instance of <see cref="OptionPriceModelResult"/> containing the theoretical
+        /// price of the specified option contract</returns>
+        public OptionPriceModelResult Evaluate(Security security, Slice slice, OptionContract contract)
+        {
+            try
+            {
+                // setting up option pricing parameters
+                var calendar = new UnitedStates();
+                var dayCounter = new Actual365Fixed();
```

Comment:
This would leak QL types into users code. We will definitely extend this point later, once other option classes are added.